### DISCUSSION TITLE
Refactoring the script

### DISCUSSION THREAD

**Discussion by sbordeyne** [ThioJoe/YT-Spammer-Purge#456] [General]
Like many others, I suppose, I've found this script thanks to Linus Tech Tips.

I've looked at the code a bit, not in too much detail yet, but it's in my opinion, a bit of a mess, which makes it hard to contribute to, and hard to implement new features when you're not acquainted with the script already.

Would you be open to a refactor from the ground up ? I've already started to organise the script, split it up in multiple files, clearly defining interfaces, adding type hints and so on, but given the size of the project, it will take a while, which means that some features might have to be backported while the refactoring is in progress.

**Comment by rachmadaniHaryono** [comment]
can i recommend using autoformatter like black, pre commit?

also on pr form there is checklist to run the test, but no documentation about how to test the program (pytest , unittest?)

i also create #459 so any developer can easily develop the program

hopefully this can be used so the program can be uploaded to pypi
